src/course-outline/data/apiHooks.ts

```diff
@@ -34,9 +34,10 @@ export const useCourseItemData = (itemId?: string, enabled: boolean = true) => (
   })
 );
 
-export const useCourseDetails = (courseId?: string) => (
+export const useCourseDetails = (courseId?: string, enabled: boolean = true) => (
   useQuery({
     queryKey: courseOutlineQueryKeys.courseDetails(courseId),
     queryFn: courseId ? () => getCourseDetails(courseId) : skipToken,
+    enabled: enabled && courseId !== undefined,
```

Comment:
```suggestion
queryFn: courseId && enabled ? () => getCourseDetails(courseId) : skipToken,
```
Using the `skipToken` if preferred while using typescript to keep type-safety when `enabled=false` ([ref](https://tanstack.com/query/latest/docs/framework/react/guides/disabling-queries#typesafe-disabling-of-queries-using-skiptoken)).